sphinx_gallery/notebook.py

```diff
@@ -57,9 +67,33 @@ def rst2md(text):
     text = re.sub(math_eq,
                   lambda match: r'$${0}$$'.format(match.group(1).strip()),
                   text)
-    inline_math = re.compile(r':math:`(.+)`')
+
+    inline_math = re.compile(r':math:`(.+?)`')
     text = re.sub(inline_math, r'$\1$', text)
 
+    directives = ('warning', 'note')
+    for directive in directives:
+        directive_re = re.compile(r'^\.\. %s::((?:.+)?(?:\n+^  .+)*)'
+                                  % directive, flags=re.M)
+        text = re.sub(directive_re,
+                      partial(directive_fun, directive=directive), text)
+
+    links = re.compile(r'^ *\.\. _.*:.*$\n', flags=re.M)
+    text = re.sub(links, '', text)
+
+    refs = re.compile(r':ref:`')
+    text = re.sub(refs, '`', text)
+
+    contents = re.compile(r'^\s*\.\. contents::.*$(\n +:\S+: *$)*\n',
+                          flags=re.M)
+    text = re.sub(contents, '', text)
+
+    images = re.compile(
+        r'^\.\. image::(.*$)(\n *:alt:(.*$))?[\n +:\S+:.*$]*', flags=re.M)
```

Comment:
Why the square brackets, did you mean to use parentheses instead in `[\n +:\S+:.*$]*`?